packages/react-components/src/components/chart/hooks/useChartsLegend.tsx

```diff
@@ -4,28 +4,66 @@ import { TableProps } from '@cloudscape-design/components/table/interfaces';
 import { GraphicComponentTextOption } from 'echarts/types/src/component/graphic/GraphicModel';
 import { SeriesOption } from 'echarts';
 import { LineSeriesOption } from 'echarts/types/src/chart/line/LineSeries';
+import { DataStream } from '@iot-app-kit/core';
+import { useHover } from 'react-use';
+
+import { borderRadiusDropdown, colorBackgroundDropdownItemHover, spaceScaledS } from '@cloudscape-design/design-tokens';
+
+import useDataStore from '../../../store';
 
 const useChartsLegend = ({
+  datastreams,
   graphic,
   series,
 }: {
+  datastreams: DataStream[];
   graphic: InternalGraphicComponentGroupOption[];
   series: SeriesOption[];
 }) => {
   const legendColumnDefinition = {
     id: 'Legends',
     header: <div className='base-chart-legend-col-header'>Properties</div>,
-    cell: (e: { lineColor: string; name: string }) => (
-      <div className='base-chart-legend-row-data-container'>
+    cell: (e: { datastream: DataStream; lineColor: string; name: string }) => {
+      const { datastream, lineColor, name } = e;
+      const highlightDataStream = useDataStore((state) => state.highlightDataStream);
```

Comment:
curious: why do we want to store the inner workings of a component in zustand/store ? this data is always used and referred within the base component.. so we could just use state ? unless this is shared across multiple basecharts..